mrmustard/physics/ansatze.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import itertools
+from abc import ABC, abstractmethod
+from typing import Any, Union
+
+import numpy as np
+from matplotlib import colors
+from matplotlib import pyplot as plt
+
+from mrmustard import math
+from mrmustard.utils.argsort import argsort_gen
+from mrmustard.utils.typing import (
+    Batch,
+    ComplexMatrix,
+    ComplexTensor,
+    ComplexVector,
+    Matrix,
+    Scalar,
+    Tensor,
+    Vector,
+)
+
+
+class Ansatz(ABC):
+    r"""Abstract parent class for Ansatze that we use to define quantum objects.
```

Comment:
I think we should use either ansatz or ansatze

Reply:
Ansatz is singular, Ansatze is plural. Where are you thinking of changing it?

Reply:
Oh lol. I did not know that